frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx

```diff
@@ -2,148 +2,129 @@
 import { Route } from "react-router";
 
 import { renderWithProviders, screen } from "__support__/ui";
-import type { State } from "metabase-types/store";
 
+import { createMockState } from "metabase-types/store/mocks";
 import {
-  createMockSettingsState,
-  createMockState,
-} from "metabase-types/store/mocks";
-import {
+  createMockDatabase,
   createMockTokenFeatures,
   createMockUser,
 } from "metabase-types/api/mocks";
 import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
 import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import type { TokenStatusStatus } from "metabase-types/api";
 
 import { AppBanner } from "./AppBanner";
 
 interface SetupOpts {
-  shouldRenderPastDueBanner: boolean;
-  shouldRenderUnpaidBanner: boolean;
-  shouldRenderDatabasePromptBanner: boolean;
+  isAdmin: boolean;
+  tokenStatusStatus: TokenStatusStatus;
+  shouldShowDatabasePromptBanner: boolean;
```

Comment:
Maybe change it and call it `hasUserDatabase` or `hasCustomDatabase` or something like that? e.g. so instead of naming setup properties based on the expected result, we call them based on the data being prepared/mocked.

Reply:
On second thought, it would be really hard to name this premise `isAdmin && isPaidPlan && !isWhiteLabeling && onlyHaveSampleDatabase` which is basically an equivalent of `shouldShowDatabasePromptBanner`.

I don't think `hasUserDatabase` would work. If that's `true` it implies we **don't** render the database prompt banner, but on the other hand,`false` doesn't guarantee the database prompt banner **would be** rendered since the 4 aforementioned conditions need to be met and this is just one of the 4 conditions.